packages/react-mutation-mapper/src/cache/DefaultStringQueryCache.ts

```diff
@@ -0,0 +1,33 @@
+import { action, observable } from 'mobx';
+
+import { CacheData, MobxCache } from '../model/MobxCache';
+
+export abstract class DefaultStringQueryCache<D>
+    implements MobxCache<D, string> {
+    protected _cache = observable.shallowMap<CacheData>();
+
+    @action
+    public get(query: string) {
+        if (!this._cache.get(query)) {
+            this._cache.set(query, { status: 'pending' });
+
+            this.fetch(query)
+                .then((d: D) =>
+                    this._cache.set(query, { status: 'complete', data: d })
+                )
+                .catch(() => this._cache.set(query, { status: 'error' }));
+        }
+
+        const data = this._cache.get(query);
+
+        return data ? data : null;
+    }
+
+    public fetch(query: string): Promise<D> {
+        return Promise.reject('No Default Implementation');
+    }
```

Comment:
Can we use abstract here so all implementations need to override this method? https://stackoverflow.com/questions/13333489/declaring-abstract-method-in-typescript

Reply:
Oh, I should have known that TS was as advanced as Java now :)